sensing/autoware_cuda_pointcloud_preprocessor/include/autoware/cuda_downsample_filter/cuda_voxel_grid_downsample_filter.hpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE__CUDA_DOWNSAMPLE_FILTER__CUDA_VOXEL_GRID_DOWNSAMPLE_FILTER_HPP_
+#define AUTOWARE__CUDA_DOWNSAMPLE_FILTER__CUDA_VOXEL_GRID_DOWNSAMPLE_FILTER_HPP_
+
+#include "autoware/cuda_downsample_filter/thrust_custom_allocator.hpp"
+
+#include <cuda_blackboard/cuda_pointcloud2.hpp>
+#include <cuda_blackboard/cuda_unique_ptr.hpp>
+
+#include <cuda_runtime.h>
+
+#include <cstdint>
+#include <iostream>
+#include <memory>
+
+#define CUDA_VGDF_ERROR_CHECK(e) (CUDA_BLACKBOARD_CHECK_CUDA_ERROR(e))
```

Comment:
nit: in autoware, should not we use the cuda_utils's version?

Reply:
same as https://github.com/autowarefoundation/autoware_universe/pull/10566#discussion_r2095511517. Thanks for your detailed review.